bob/src/cleaner.rs

```diff
@@ -22,27 +24,74 @@ impl Cleaner {
             hard_open_blobs,
             bloom_filter_memory_limit,
             index_memory_limit,
+            index_cleanup_requested: Mutex::new(false),
         }
     }
 
-    pub(crate) fn spawn_task(&self, backend: Arc<Backend>) {
+    pub(crate) fn spawn_task(&self, cleaner: Arc<Cleaner>, backend: Arc<Backend>) {
         tokio::spawn(Self::task(
-            backend,
+            cleaner.clone(),
+            backend.clone(),
             self.old_blobs_check_timeout,
             self.soft_open_blobs,
             self.hard_open_blobs,
             self.bloom_filter_memory_limit,
-            self.index_memory_limit,
         ));
+        tokio::spawn(Self::fast_cleaner_task(
+            cleaner,
+            backend,
+            Duration::from_secs(30),
+            self.index_memory_limit
+        ));
+    }
+
+    pub(crate) async fn request_index_cleanup(&self) {
+        let mut lock = self.index_cleanup_requested.lock().await;
+        *lock = true;
+    }
+
+    async fn fast_cleaner_task(
+        cleaner: Arc<Cleaner>,
+        backend: Arc<Backend>,
+        t: Duration,
+        index_memory_limit: Option<usize>,
+    ) {
+        let mut interval = interval(t);
+        loop {
+            interval.tick().await;
+            if let Some(limit) = index_memory_limit {
+                let mut should_clean_up = false;
+                {
+                    let mut lock = cleaner.index_cleanup_requested.lock().await;
```

Comment:
`Notify` is better for this scenario: https://docs.rs/tokio/1.9.0/tokio/sync/struct.Notify.html
`notify_waiters` can be used to unlock the task: https://docs.rs/tokio/1.9.0/tokio/sync/struct.Notify.html#method.notify_waiters
After the notification is received, an additional small delay can be added to prevent immediate cleanup